network-api/networkapi/wagtailpages/migrations/0163_remove_focusarea_page_homepage_hero_intro_body_and_more.py

```diff
@@ -0,0 +1,45 @@
+# Generated by Django 4.2.15 on 2024-09-19 20:09
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```

Comment:
To respond to @mmmavis and @robdivincenzo 's [comments](https://github.com/MozillaFoundation/foundation.mozilla.org/pull/12858#discussion_r1757640739):

After looking through the changes made, I do not believe that separate deployments are needed, and we are good to go with the auto-generated migration files in a single PR.


However, I can see what they mean when they mention separating the changes to the focus area and homepage models in separate files.


The easiest way to do so would be:

1. Run `inv new-db`
2. Remove the `page` field from the `FocusArea` model
3. Run `inv makemigrations` and rename the generated migration file to something like "0xxx_remove_page_from_focusarea.py"
5. Add the new fields to the `HomePage` model
6. Run `inv makemigrations` again and rename the file to something like "0xxx_update_homepage_hero_fields.py"

Regarding the need to populate the new fields, I believe the safest approach is to add a post-migration operation. This would ensure that the template remains intact without waiting for someone to manually populate the fields after deploying.


To do so, you can update the migration file generated in step 6 to something like:


```python
from django.db import migrations
...

def populate_hero_intro_fields(apps, schema_editor):
    Homepage = apps.get_model("wagtailpages", "Homepage")

    link_value = {
        "link_to": "external_url",
        "external_url": "<link_button_url>", <-- The URL sent to us from Design
        "label": "<link_button_label>", <-- The Label sent to us from Design
        "new_window": True,
    }

    for homepage in Homepage.objects.all():
        homepage.hero_intro_heading = "<intro_heading_text>" <-- The heading text sent to us from Design
        homepage.hero_intro_body = "<intro_body_text>" <-- The body text sent to us from Design

        # Create the StreamField value for hero_intro_link
        homepage.hero_intro_link = [("link", link_value)]

        # Save changes
        homepage.save()


class Migration(migrations.Migration):

    dependencies = [
...
    ]

    operations = [
... 
        migrations.RunPython(populate_hero_intro_fields),
    ]
```


Once this is deployed, the only remaining task will be to submit the homepage for translation. In the meantime, the translated pages will default to English until the translations are ready. 👍